src/crl.rs

```diff
@@ -220,27 +243,41 @@ impl<'a> BorrowedCertRevocationList<'a> {
             }
         })
     }
+}
 
-    /// Try to find a [`RevokedCert`] in the CRL that has a serial number matching `serial`.
-    /// This may yield an error if the CRL has malformed revoked certificates.
-    pub fn find_serial(&self, serial: &[u8]) -> Result<Option<RevokedCert>, Error> {
-        // TODO(XXX): This linear scan is sub-optimal from a performance perspective, but avoids
-        //            any allocation. It would be nice to offer a speedier alternative for
-        //            when the alloc feature is enabled:
-        //            https://github.com/rustls/webpki/issues/80
-        self.into_iter()
-            .find_map(|revoked_cert_result| match revoked_cert_result {
-                Err(e) => Some(Err(e)),
-                Ok(revoked_cert) if revoked_cert.serial_number.eq(serial) => Some(Ok(revoked_cert)),
-                _ => None,
-            })
-            .transpose()
-    }
+impl Sealed for BorrowedCertRevocationList<'_> {}
 
-    /// Raw DER encoding of the issuer of the CRL.
-    pub fn issuer(&self) -> &[u8] {
+impl CertRevocationList for BorrowedCertRevocationList<'_> {
+    fn issuer(&self) -> &[u8] {
         self.issuer.as_slice_less_safe()
     }
+
+    fn find_serial(&self, serial: &[u8]) -> Result<Option<RevokedCert>, Error> {
+        for revoked_cert_result in self.into_iter() {
```

Comment:
You probably don't need `into_iter()` here? That's the interface that the language's `for` keyword relies on anyway, so it should call it for you.

Reply:
You're right, thanks :+1: